agentops/helpers.py

```diff
@@ -18,18 +18,38 @@
     "crewai": (False, True),
 }
 
+ao_instances = {}
+
 
 def singleton(class_):
-    instances = {}
 
     def getinstance(*args, **kwargs):
-        if class_ not in instances:
-            instances[class_] = class_(*args, **kwargs)
-        return instances[class_]
+        if class_ not in ao_instances:
+            ao_instances[class_] = class_(*args, **kwargs)
+        return ao_instances[class_]
+
+    return getinstance
+
+
+def conditional_singleton(class_):
+
+    def getinstance(*args, **kwargs):
+        use_singleton = kwargs.pop("use_singleton", True)
+        if use_singleton:
+            if class_ not in ao_instances:
+                ao_instances[class_] = class_(*args, **kwargs)
+            return ao_instances[class_]
+        else:
+            return class_(*args, **kwargs)
 
     return getinstance
 
 
+def clear_singletons():
+    global ao_instances
```

Comment:
Can we check if this is where global goes or if it should be on line 21 where we define ao_instances?

Reply:
you put global before its use